### PR TITLE
lookup: reject duplicate table names

### LINKED ISSUE #5316: rsyslogd: a lookup table could not be initialized: failed at init-step 4
### Expected behavior

When loading a large (and valid) configuration file with many lookup tables with `-N1`, the checks should return a valid configuration.

### Actual behavior

(Tested on an HardenedBSD 13-stable, with 1500 allowed threads-per-process with the `kern.threads.max_threads_per_proc` parameter)

When trying to validate the configuration, the run will return errors:
`rsyslogd: a lookup table could not be initialized: failed at init-step 4 (please enable debug logs for details): Resource temporarily unavailable [v8.2206.0 try https://www.rsyslog.com/e/2428 ]
rsyslogd: a lookup table could not be initialized: failed at init-step 4 (please enable debug logs for details): Resource temporarily unavailable [v8.2206.0 try https://www.rsyslog.com/e/2428 ]
rsyslogd: a lookup table could not be initialized: failed at init-step 4 (please enable debug logs for details): Resource temporarily unavailable [v8.2206.0 try https://www.rsyslog.com/e/2428 ]
rsyslogd: a lookup table could not be initialized: failed at init-step 4 (please enable debug logs for details): Resource temporarily unavailable [v8.2206.0 try https://www.rsyslog.com/e/2428 ]
[...]
rsyslogd: error during parsing file ./rsyslog_test.conf, on or before line 126544: lookup table 'lookup_1' not found (used in function: lookup) [v8.2206.0 try https://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file ./rsyslog_test.conf, on or before line 126552: lookup table 'lookup_2' not found (used in function: lookup) [v8.2206.0 try https://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file ./rsyslog_test.conf, on or before line 126553: lookup table 'lookup_3' not found (used in function: lookup) [v8.2206.0 try https://www.rsyslog.com/e/2207 ]
rsyslogd: error during parsing file ./rsyslog_test.conf, on or before line 126555: lookup table 'lookup_4' not found (used in function: lookup) [v8.2206.0 try https://www.rsyslog.com/e/2207 ]
`

### Steps to reproduce the behavior

This is very environment-dependent, but should be reproduced with those conditions:
1. check the kernel allowed thread per process (in my case, kern.threads.max_threads_per_proc = 1500)
2. generate a configuration file with more lookup_table() definitions than the allowed number of thread per process
3. try to validate the configuration with rsyslogd -f configuration.conf -N1

### Environment
- rsyslog version: 
- platform: 
unfortunately, I cannot provide an exemple configuration file for this error

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary

- reject duplicate `lookup_table` names before allocating another table entry
- keep existing invalid-initial-table registration behavior so HUP reload can recover
- add RainerScript and YAML regression tests for duplicate lookup table names

## Validation

- `make -j$(nproc) check TESTS=""`
- `./tests/lookup_table_duplicate.sh`
- `./tests/yaml-lookup-table-duplicate.sh`
- `./tests/lookup_table_bad_configs.sh`
- `./tests/lookup_table_bad_configs-vg.sh`
- `./tests/array_lookup_table_misuse-vg.sh`
- `./tests/lookup_table.sh`
- `./tests/multiple_lookup_tables.sh`
- `make distcheck TEST_RUN_TYPE=MOCK-OK -j$(nproc)`
- Ubuntu 26.04 static analyzer container: no bugs found
- Ubuntu 26.04 container CI: 1259 total, 1231 pass, 27 skip, 1 fail
  - `imhttp-post-payload-multi-vg.sh` failed because CivetWeb `mg_start` failed
  - direct rerun of `imhttp-post-payload-multi-vg.sh` in the same container image passed

closes https://github.com/rsyslog/rsyslog/issues/5316

### CODE REVIEW

File: runtime/lookup.c [hunk 1]
```diff
@@ -1105,30 +1114,40 @@ rsRetVal lookupTableDefProcessCnf(struct cnfobj *o) {
...
+    if (name == NULL || filename == NULL) {
         iRet = RS_RET_INTERNAL_ERROR;
-        LogError(0, iRet, "internal error: lookup table name not set albeit being mandatory");
+        LogError(0, iRet, "internal error: lookup table name or file not set albeit being mandatory");
         ABORT_FINALIZE(iRet);
     }
+    if (lookupFindTable(name) != NULL) {
+        LogError(0, RS_RET_CONFIG_ERROR, "lookup_table: duplicate name '%s' in current config set", name);
+        ABORT_FINALIZE(RS_RET_CONFIG_ERROR);
+    }
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply runtime/lookup.c:1148]: Done.
